src/components/editor/Store/create.ts

```diff
@@ -105,18 +124,58 @@ const getInitialState = <T>(
             );
         },
 
-        setStatus: (newVal) => {
+        setStatus: (value, path) => {
             set(
                 produce((state) => {
-                    state.isSaving = newVal === EditorStatus.SAVING;
-                    state.isEditing = newVal === EditorStatus.EDITING;
-                    state.status = newVal;
+                    if (Object.hasOwn(state.statuses, path)) {
+                        state.statuses[path] = value;
+                    } else {
+                        state.statuses = { ...state.statuses, [path]: value };
+                    }
+
+                    const editorStatuses: EditorStatus[] = Object.values(
+                        state.statuses
+                    );
+
+                    state.isSaving = editorStatuses.some(
+                        (status) => status === EditorStatus.SAVING
+                    );
+                    state.isEditing = editorStatuses.some(
+                        (status) => status === EditorStatus.EDITING
+                    );
+
+                    state.invalidEditors = Object.entries(state.statuses)
+                        .filter(
+                            ([_key, status]) =>
+                                status === EditorStatus.INVALID ||
+                                status === EditorStatus.SAVE_FAILED ||
+                                status === EditorStatus.OUT_OF_SYNC
+                        )
+                        .map(([key, _status]) => key);
```

Comment:
Thinking out loud - this one wouldn't be as easy to pull off but we should probably make some kind of reusable store helper/extension/something to handle status checking and make it work for singular or multiples like here.